File: MyApplication/library/src/main/java/com/a65apps/kalimullinilnazrafilovich/library/applicaiton/fragments/ContactDetailsFragment.java
package com.a65apps.kalimullinilnazrafilovich.library.applicaiton.fragments;

import android.app.Application;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo;
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.delegate.details.ButtonsDelegate;
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.delegate.details.ViewsDetailsDelegate;
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.di.interfaces.ContactDetailsContainer;
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.di.interfaces.HasAppContainer;
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.presenters.ContactDetailsPresenter;
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.views.ContactDetailsView;
import com.a65apps.kalimullinilnazrafilovich.myapplication.R;

import java.util.Objects;

import javax.inject.Inject;
import javax.inject.Provider;

import moxy.MvpAppCompatFragment;
import moxy.presenter.InjectPresenter;
import moxy.presenter.ProvidePresenter;

public class ContactDetailsFragment extends MvpAppCompatFragment implements ContactDetailsView {

    @InjectPresenter
    @NonNull
    public ContactDetailsPresenter contactDetailsPresenter;
    @Inject
    @NonNull
    public Provider<ContactDetailsPresenter> contactDetailsPresenterProvider;

    private ViewsDetailsDelegate viewsDetailsDelegate;
    private ButtonsDelegate buttonsDelegate;

    private String id;

    @NonNull
    public static ContactDetailsFragment newInstance(@NonNull String id) {
        ContactDetailsFragment fragment = new ContactDetailsFragment();
        Bundle args = new Bundle();
        args.putString("id", id);
        fragment.setArguments(args);
        return fragment;
    }

    @NonNull
    @ProvidePresenter
    ContactDetailsPresenter providePresenter() {
        return contactDetailsPresenterProvider.get();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        Application app = requireActivity().getApplication();
        if (!(app instanceof HasAppContainer)) {
            throw new IllegalStateException();
        }
        ContactDetailsContainer contactDetailsComponent = ((HasAppContainer) app).appContainer()
                .plusContactDetailsContainer();

        contactDetailsComponent.inject(this);

        super.onAttach(context);
    }

    @Override
    @NonNull
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_contact_details, container, false);
        Objects.requireNonNull(getActivity()).setTitle(getString(R.string.title_toolbar_contact_details));

        viewsDetailsDelegate = new ViewsDetailsDelegate(view);
        buttonsDelegate = new ButtonsDelegate(view, getActivity(), contactDetailsPresenter);

        buttonsDelegate.clickButtons();

        id = getArguments().getString("id");

        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        contactDetailsPresenter.showDetails(id);
    }

    @Override
    public void showContactDetail(@NonNull ContactDetailsInfo contactDetailsInfo) {
        viewsDetailsDelegate.setViews(contactDetailsInfo);
        buttonsDelegate.setStatusButtons(contactDetailsInfo);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        viewsDetailsDelegate.unBind();
        buttonsDelegate.unBind();
    }
}

File: library/src/main/java/com/a65apps/kalimullinilnazrafilovich/library/applicaiton/fragments/ContactDetailsFragment.kt
package com.a65apps.kalimullinilnazrafilovich.library.applicaiton.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.lifecycle.LiveData
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.delegate.details.ButtonsDelegate
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.delegate.details.ViewsDetailsDelegate
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.di.interfaces.HasAppContainer
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.viewModels.ContactDetailsViewModel
import com.a65apps.kalimullinilnazrafilovich.library.applicaiton.viewModels.factory.ContactDetailsFactory
import com.a65apps.kalimullinilnazrafilovich.myapplication.R
import javax.inject.Inject

class ContactDetailsFragment : Fragment() {
    @Inject
    lateinit var viewModelFactory: ContactDetailsFactory
    private lateinit var viewModel: ContactDetailsViewModel

    private lateinit var viewsDetailsDelegate: ViewsDetailsDelegate
    private lateinit var buttonsDelegate: ButtonsDelegate

    private val id by lazy {
        requireArguments().getString("id")
            ?: throw IllegalArgumentException("ContactDetailsFragment required 'id' argument")
    }

    companion object {
        @JvmStatic
        fun newInstance(id: String?) = ContactDetailsFragment().apply {
            arguments = bundleOf(
                "id" to id
            )
        }
    }

    override fun onAttach(context: Context) {
        val app = requireActivity().application
        if (app !is HasAppContainer) {
            throw IllegalStateException()
        }
        val contactDetailsComponent = app.appContainer().plusContactDetailsContainer()

        contactDetailsComponent.inject(this)

        super.onAttach(context)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        viewModel =
            ViewModelProvider(this, viewModelFactory).get(ContactDetailsViewModel::class.java)
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_contact_details, container, false)
        activity?.title = getString(R.string.title_toolbar_contact_details)

        viewsDetailsDelegate = ViewsDetailsDelegate(view)
        buttonsDelegate = ButtonsDelegate(view, requireActivity(), viewModel)

        buttonsDelegate.clickButtons()

        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val details: LiveData<ContactDetailsInfo> = viewModel.getContactDetails(id)
        details.observe(
            viewLifecycleOwner,
            Observer { contact ->
                viewsDetailsDelegate.setViews(contact)
                buttonsDelegate.setStatusButtons(contact, viewLifecycleOwner)
            })
    }

    override fun onDestroyView() {
        viewsDetailsDelegate.unBind()
        buttonsDelegate.unBind()
        super.onDestroyView()
    }
}
